### ISSUE TITLE
--branches flag does not support prereleases

### ISSUE THREAD

**Issue by bdepalo** [semantic-release/semantic-release#3463]
### Current behavior

`semantic-release --branches '{"name":"main"},{"name":"version-test","prerelease":"snap"}' --dry-run --debug`

Splits `{"name":"version-test","prerelease":"snap"}` into two individual branch objects

### Expected behavior

 `{"name":"version-test","prerelease":"snap"}` remains a single object despite the `,` being present (as required)
 
 Relevant line of code doing parsing: https://github.com/semantic-release/semantic-release/blob/00fc6811fb756b24e7215f386d121098d6cf6da8/cli.js#L12

### `semantic-release` version

24.1.2

### CI environment

GitLab

### Plugins used

Not relevant

### `semantic-release` configuration

Not relevant

### CI logs

Relevant log information:

2024-10-02T14:41:46.560Z semantic-release:config options values: {
  branches: [
    '{"name":"main"}',
    '{"name":"version-test"',
    '"prerelease":"snap"}'
  ],
  repositoryUrl: 'REDACTED),
  tagFormat: 'v${version}',
  plugins: [
    [ '@semantic-release/commit-analyzer', [Object] ],
    [ '@semantic-release/release-notes-generator', [Object] ],
    [ '@semantic-release/exec', [Object] ],
    [ '@semantic-release/gitlab', [Object] ]
  ],
  _: [],
  b: [
    '{"name":"main"}',
    '{"name":"version-test"',
    '"prerelease":"snap"}'
  ],
  'dry-run': true,
  d: true,
  dryRun: true,
  debug: true,
  '$0': 'semantic-release'
}

**Comment by babblebey** [comment]
This I can confirm to be an issue indeed, BUT you there's something you could've done to avoid this.... I'll tell ya 😉 

Though the thing is, the `--branches` and other arguments that accepts array of object in semantic-release configuration, do not exactly have proper parsing of their value whenever there's an object present in the data at my look of it.... the parser splits/handles the value as though its just an array of strings separated with commas `,`... Maybe we can fix that for sure YES. 👍 

But, here what you can do.... If you were to have used the value `{"name": "main"}` alone in you `--branches` config, it wouldn't have worked either, you still get the `'ERELEASEBRANCHES'` error... You can fight this by configuring your `main` branch writing it directly as a string instead of using the object with a `name` property. Also, it wouldn't hurt to just write a `.releaserc` file to just get going for this 😉

**Comment by bdepalo** [comment]
yeah i ended up doing a `.releaserc` to resolve the issue. this is really only a "problem" because I am trying to both prerelease and release from the same branch (another story for another day lol)
